### PR TITLE
new package: py-uncertainties

### PR DESCRIPTION
**Pull request by Sinan81**
A dependency of #16975

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-uncertainties/package.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    url      = "https://pypi.io/packages/source/u/uncertainties/uncertainties-3.1.4.tar.gz"
+
+    version('3.1.4', sha256='63548a94899f2a51eeb89b640f6ac311f481a8016b37dce157186e44619bc968')
+
+    variant('numpy', default=False, description='Enable extra features involving numpy')
+    variant('docs',  default=False, description='Build with documentation')
+
+    depends_on('python@2.7:',   type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-future',     type=('build', 'run'), when='python@:2')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/py-uncertainties/package.py]: (EDIT) This needs to be `when='^python@:2'`

File: var/spack/repos/builtin/packages/py-uncertainties/package.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    url      = "https://pypi.io/packages/source/u/uncertainties/uncertainties-3.1.4.tar.gz"
+
+    version('3.1.4', sha256='63548a94899f2a51eeb89b640f6ac311f481a8016b37dce157186e44619bc968')
+
+    variant('numpy', default=False, description='Enable extra features involving numpy')
+    variant('docs',  default=False, description='Build with documentation')
+
+    depends_on('python@2.7:',   type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-future',     type=('build', 'run'), when='^python@:2')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-uncertainties/package.py]:
```suggestion
    depends_on('py-future',     type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-uncertainties/package.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    depends_on('py-numpy',      type=('build', 'run'), when='+numpy')
+    depends_on('py-sphinx',     type=('build', 'run'), when='+docs')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-uncertainties/package.py]: These two are both optional, I would either remove them or add variants for `+optional` and `+docs` and make them default to False

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-uncertainties/package.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    version('3.1.4', sha256='63548a94899f2a51eeb89b640f6ac311f481a8016b37dce157186e44619bc968')
+
+    variant('optional', default=False, description='Enable extra features involving numpy')
+    variant('docs',  default=False, description='Build with documentation')
+
+    depends_on('python@2.7:',   type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-future',     type=('build', 'run'))
+    depends_on('py-numpy',      type=('build', 'run'), when='+optional')
+    depends_on('py-sphinx',     type=('build', 'run'), when='+docs')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-uncertainties/package.py]:
```suggestion
    depends_on('py-sphinx',     type='build', when='+docs')
```

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**